src/onemod/fsutils/data_loader.py

```diff
@@ -0,0 +1,77 @@
+from pathlib import Path
+from typing import Literal
+
+import pandas as pd
+import polars as pl
+
+from onemod.fsutils.io import DataIO, dataio_dict
+
+
+class DataLoader:
+    """Handles loading and dumping of data files with optional lazy loading,
+    column selection, and subset filtering."""
+
+    io_dict: dict[str, DataIO] = dataio_dict
+
+    def load(
+        self,
+        path: Path,
+        lazy: bool = False,
+        backend: Literal["polars", "pandas"] = "polars",
```

Comment:
One random idea is that, we can combine arguments `lazy` and `backend` to
```python
backend: Literal["polars", "polars_lazy", "polars_eager", "pandas"] = "polars"
```
where `"polars"` is an alias as `"polars_lazy"`, this way we can reduce a argument, and don't have to worry about user input `pandas` and the backend and `lazy=True`.